File: Level1/220822/Java/KthNumber.java
import java.util.Arrays;

public class KthNumber {
  public int[] solution(int[] array, int[][] commands) {
    int[] answer = new int[commands.length];

    for (int i = 0; i < commands.length; i += 1) {
      int[] slicingArray = slice(array, commands[i]);

      Arrays.sort(slicingArray);

      answer[i] = slicingArray[commands[i][2] - 1];
    }

    return answer;
  }

  public int[] slice(int[] array, int[] commands) {
    int[] slicingArray;
    int startIndex = commands[0];
    int endIndex = commands[1];

    slicingArray = Arrays.copyOfRange(array, startIndex - 1, endIndex);
    return slicingArray;
  }
}

File: Programmers/Level1/220823/JavaScript/KthNumber.js
// eslint-disable-next-line import/prefer-default-export
export function solution(array, commands) {
  const answer = [commands.length];

  for (let i = 0; i < commands.length; i += 1) {
    const startIndex = commands[i][0];
    const endIndex = commands[i][1];

    const slicingArray = array.slice(startIndex - 1, endIndex);
    slicingArray.sort((a, b) => a - b);

    answer[i] = slicingArray[commands[i][2] - 1];
  }

  return answer;
}
